vespa-feed-client/src/main/java/ai/vespa/feed/client/impl/ApacheCluster.java

```diff
@@ -86,7 +89,8 @@ public void dispatch(HttpRequest wrapped, CompletableFuture<HttpResponse> vessel
                 SimpleHttpRequest request = new SimpleHttpRequest(wrapped.method(), wrapped.path());
                 request.setScheme(endpoint.url.getScheme());
                 request.setAuthority(new URIAuthority(endpoint.url.getHost(), portOf(endpoint.url)));
-                request.setConfig(requestConfig);
+                long timeoutMillis = wrapped.timeout() == null ? 190_000 : wrapped.timeout().toMillis() * 11 / 10 + 1_000;
+                request.setConfig(requestConfig.setResponseTimeout(Timeout.ofMilliseconds(timeoutMillis)).build());
```

Comment:
Thread-safe to mutate shared request config builder?

Reply:
Right ... this used to be single-threaded dispatch :)